mailpoet/lib/Segments/SegmentSubscribersRepository.php

```
@@ -47,6 +47,63 @@ public function getSubscriberIdsInSegment(int $segmentId): array {
+  /**
+   * Fetch a batch of unique subscriber IDs across multiple segments using cursor-based pagination.
+   * Uses SQL-level DISTINCT, ORDER BY, and LIMIT to avoid loading all IDs into memory.
+   *
+   * @param int[] $segmentIds
+   * @param int   $afterId  Cursor: only return subscribers with ID > this value
+   * @param int   $limit    Maximum number of IDs to return
+   * @return int[]
+   */
+  public function getSubscriberIdsInSegments(array $segmentIds, int $afterId = 0, int $limit = 100): array {
+    $subscribersTable = $this->entityManager->getClassMetadata(SubscriberEntity::class)->getTableName();
+    $subscribersSegmentsTable = $this->entityManager->getClassMetadata(SubscriberSegmentEntity::class)->getTableName();
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$subscribersSegmentsTable'. (undefined)

(UnusedLocalVariable)